### PR TITLE
Java code tabs are first in index. Close #46

### LINKED ISSUE #46: Java code tabs should be first
Java should be the first and default tab for all code blocks because it is the most popular language by a significant amount.

### PR DESCRIPTION
**Pull request by JamieSinn**
Moved any Java tabs that were after CPP and now are above.

### CODE REVIEW

File: source/docs/software/actuators/solenoids.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+be passed to the DoubleSolenoid if teams have a non-standard PCM CAN ID
+
+.. tabs::
+
+   .. code-tab:: java
+
+        // Using "import static an.enum.or.constants.inner.class.*;" helps reduce verbosity
+        // this replaces "DoubleSolenoid.Value.kForward" with just kForward
+        // further reading is avalible at https://www.geeksforgeeks.org/static-import-java/
+        import static edu.wpi.first.wpilibj.DoubleSolenoid.Value.*;
```
Comment (AustinShalit) [inline source/docs/software/actuators/solenoids.rst]: This is contrary to the WPILib style guide.  This should be changed to match the guide.

File: source/docs/software/actuators/solenoids.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
+Operating pneumatic cylinders
+=============================
+
+Solenoid control
+----------------
+
+FRC teams use solenoids to preform a variety of tasks, from shifting
+gearboxes to operating robot mechanisms. A solenoid is a valve used to
+electronically switch a pressurised air line "on" or "off". For more
```
Comment (AustinShalit) [inline source/docs/software/actuators/solenoids.rst]: Not completely true.  We do have and allow electric solenoid actuators.  This sentence should clarify this distinction.

File: source/docs/software/actuators/solenoids.rst [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+information on solenoids, see `this wikipedia
+article <https://en.wikipedia.org/wiki/Solenoid_valve>`__. Solenoids are
+controlled by a robot's Pneumatics Control Module, or PCM, which is in
+turn connected to the robot's RoboRIO via CAN. The easiest way to see a
```
Comment (AustinShalit) [inline source/docs/software/actuators/solenoids.rst]:
```suggestion
turn connected to the robot's roboRIO via CAN. The easiest way to see a
```

File: source/docs/software/actuators/solenoids.rst [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+solenoid's state is via the small red LED (which indicates if the valve
+is "on" or not), and solenoids can be manually actuated when un-powered
+with the small button adjacent to the LED.
```
Comment (AustinShalit) [inline source/docs/software/actuators/solenoids.rst]: This should clarify that the button is on the solenoid, not the PCM.

File: source/docs/software/actuators/solenoids.rst [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+Single acting solenoids apply or vent pressure from a single output
+port. They are typically used either when an external force will provide
+the return action of the cylinder (spring, gravity, separate mechanism)
+or in pairs to act as a double solenoid. A double solenoid switches air
```
Comment (AustinShalit) [inline source/docs/software/actuators/solenoids.rst]: The photo below describes the 2 kinds of solenoids that have ever been included in the KOP. 

FRC Solenoids

The image on the left is what FRC calls "Single" solenoids; on the right is "Double".  As you can see both devices have 2 output ports (labeled 2 and 4).  The primary difference between single and double solenoids is what happens when they are not powered (Like between auto and teleop or at the end of a match).  Double solenoids will stay in their last commanded state.  Single solenoids will go to their default.

File: source/docs/software/actuators/solenoids.rst [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+Single Solenoids in WPILib
+--------------------------
+
+Using a single acting solenoid
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Single solenoids in WPILib are controlled using the Solenoid class. To
+construct a Solenoid object, simply pass the desired port number
+(assumes Node ID 0) or Node ID and port number to the constructor. To
+set the value of the solenoid call set(true) to enable or set(false) to
```
Comment (AustinShalit) [inline source/docs/software/actuators/solenoids.rst]: Wrap with in line code blocks.

File: source/docs/software/actuators/solenoids.rst (hunk 5 above)
Comment (AustinShalit) [reply source/docs/software/actuators/solenoids.rst]: Correction.  One other valve was included (I think before 2010).  It was a  VPLE18-M5H-4/2-1/4

valve symbols_42_mono-stable_2 port designations_en

It is the same as the current single solenoids but the two sides of the system share a single exhaust port.

File: source/docs/software/actuators/solenoids.rst (hunk 2 above)
Comment (JamieSinn) [reply source/docs/software/actuators/solenoids.rst]: Adding in a short blurb about this being focused on pneumatics, but the electrical solenoids are legal.

**Pull request merged**